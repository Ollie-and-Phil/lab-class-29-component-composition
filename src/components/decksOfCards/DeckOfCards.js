import React from 'react';
import PropTypes from 'prop-types';
import Card from '../cards/Card';
import Deck from '../decks/Deck';

const DeckOfCards = ({ cards }) => {
  const cardElements = cards.map(card => (
    <div key={`${card._id}`}>
      <Card title={card.name} image={card.image || card.gifLink} />
    </div>
  ));

  return (
    <Deck>
      {cardElements}
    </Deck>
  );
};

DeckOfCards.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    image: PropTypes.string,
    gifLink: PropTypes.string,
    _id: PropTypes.string.isRequired
  }))
};

export default DeckOfCards;
