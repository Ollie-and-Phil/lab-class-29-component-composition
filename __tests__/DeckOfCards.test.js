import React from 'react';
import { shallow } from 'enzyme';
import DeckOfCards from '../src/components/decksOfCards/DeckOfCards';

describe('DeckOfCards', () => {
  it('renders a deck of cards', () => {
    const characters = [{ name: 'Ollie', image: 'Ollie.png', _id:'you' }, { name: 'Ryan', image: 'Ryan.txt', _id:'sir' }];
    const wrapper = shallow(<DeckOfCards cards={characters}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a deck of gifCards', () => {
    const gifs = [{ gifLink: 'Ollie.png', _id:'you' }, { gifLink: 'Ryan.txt', _id:'sir' }];
    const wrapper = shallow(<DeckOfCards cards={gifs}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
