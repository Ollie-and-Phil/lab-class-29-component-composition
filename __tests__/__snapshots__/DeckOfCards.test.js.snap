// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckOfCards renders a deck of cards 1`] = `
<Deck>
  <div
    key="you"
  >
    <Card
      image="Ollie.png"
      title="Ollie"
    />
  </div>
  <div
    key="sir"
  >
    <Card
      image="Ryan.txt"
      title="Ryan"
    />
  </div>
</Deck>
`;

exports[`DeckOfCards renders a deck of gifCards 1`] = `
<Deck>
  <div
    key="you"
  >
    <Card
      image="Ollie.png"
    />
  </div>
  <div
    key="sir"
  >
    <Card
      image="Ryan.txt"
    />
  </div>
</Deck>
`;
